mlflow/models/signature.py

```diff
@@ -329,10 +338,16 @@ def _infer_signature_from_input_example(
         `wrapped_model`.
     """
     try:
-        input_ex = _Example(input_example).inference_data
-        input_schema = _infer_schema(input_ex)
+        # We shouldn't do this at least for lists --> it produces wrong results
+        # ["input"] --> the output pandas dataframe contains column names in the signature
+        # ["input1", "input2", "input3"] --> signature would become [0: string, 1: string, 2: string]
+        input_example = _Example(input_example)
         # Copy the input example so that it is not mutated by predict()
-        prediction = wrapped_model.predict(deepcopy(input_ex))
+        input_ex = deepcopy(input_example.inference_data)
+        params = input_example.inference_params
+        input_schema = _infer_schema(input_ex)
+        params_schema = _infer_param_schema(params) if params else None
+        prediction = wrapped_model.predict(input_ex, params=params)
```

Comment:
Problems here. I think we shouldn't replace input_example with _Example for inferring signature. It's not consistent.